cores/esp8266/core_esp8266_i2s.c

```diff
@@ -194,46 +219,94 @@ bool ICACHE_FLASH_ATTR i2s_write_lr(int16_t left, int16_t right){
   return i2s_write_sample(sample);
 }
 
+bool ICACHE_FLASH_ATTR i2s_write_lr_nb(int16_t left, int16_t right){
+  int sample = right & 0xFFFF;
```

Comment:
`i2s_write_lr()` does use int32_t, I copied the code and only changed that `i2s_write_sample(sample);` to `i2s_write_sample_nb(sample)`;